### PR TITLE
refactor login

### PR DESCRIPTION
**Pull request by thetif**
This is a very big refactor. Previously, all of the login was handled with if statements in in the Login component. I took it out and make a login router. 

issues #2632 #2648 #2706 #2864 

### This pull request changes...

- Moved some of the navigation that was being handled in the Login component into a LoginRouter
- Removed some of the redux values and dispatch types that were no longer needed
- Updated some tests and components to reduce console warnings.
- Functionality shouldn't change considerably
- Removes "Cancel" from state affiliation selection
- Resolves the logout issue where user is not fully logged out
- Combines the MFA selection into a single option

### This pull request is ready to merge when...

- [x] Tests have been updated (and all tests are passing)
- [x] This code has been reviewed by someone other than the original author
- [ ] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented
- [x] The change has been documented
- [x] Associated OpenAPI documentation has been updated
- [x] Changelog is updated as appropriate

### This feature is done when...

- [ ] QA has approved the experience
- [ ] Design has approved the experience
- [x] Product has approved the experience

### CODE REVIEW

File: web/src/containers/LoginPageRoutes.test.js
Comment (radavis) [inline web/src/containers/LoginPageRoutes.test.js:51]: Just a visual nitpic: You could define `initialHistory` first, and avoid spreading the `renderOptions` object over three lines:

```
const initialHistory = ['/login/locked-out'];
const {
  queryByRole
} = renderWithConnection(
  <LoginPageRoutes {...props} useRouteMatch={useRouteMatch} />, { initialHistory }
);
```

Prettier might possibly wreck this formatting though. So, just a thought. :man_shrugging:

File: web/src/util/auth.js
Comment (radavis) [inline web/src/util/auth.js:179]: I'm curious why we have implementing our own 'cookie string to object' code in this app. It might be fine, but we should weight the pros and cons of using an external library to do this work. 

https://openbase.com/categories/js/best-nodejs-cookies-libraries?orderBy=RECOMMENDED&

**Review by radavis — COMMENTED** [review]
Looks good. Will approve after tomorrow's review.

Note: Add `window.scroll = () => {}` to `setup.enzyme.test.js` to get rid of the `window.scroll` errors when testing.

File: web/src/containers/LoginPageRoutes.test.js
Comment (thetif) [reply web/src/containers/LoginPageRoutes.test.js:51]: Yeah I was looking for an example of that better component set up that you've been doing, but I couldn't find one.

**Review by tbolt — COMMENTED** [review]
Code looks good. I believe there is an issue with the style on the login pages-
loginstyleissue

**Review by tbolt — APPROVED** [review]
The code looks good, ill approve it so others can take a look. 

There are a couple style issues im seeing. I can take these issues on in another ticket if you want.

I am seeing that the login page is showing the red alert state for the password when I haven't submitted the form-
image

The "Verify Your Identity" page is missing the grey background like the other login pages.

File: web/src/components/ConsentBanner.test.js
Comment (eStar386) [inline web/src/components/ConsentBanner.test.js:44]: @thetif Will these type of accessibility checks be added to all development tests in the future.

Comment (thetif) [reply web/src/components/ConsentBanner.test.js:44]: yes that's the plan

**Review by radavis — APPROVED** [review]
:+1:

File: documentation/sequence-diagram-authorization.md
Comment (knollfear) [inline documentation/sequence-diagram-authorization.md:4]: I think if you swap server and Okta in this list the diagram looks better.  You can render this diagram here to see  https://mermaid-js.github.io/mermaid-live-editor/#/edit/eyJjb2RlIjoic2VxdWVuY2VEaWFncmFtXG4gIHBhcnRpY2lwYW50IFcgYXMgV2ViIEFwcFxuICBwYXJ0aWNpcGFudCBTIGFzIFNlcnZlclxuICBwYXJ0aWNpcGFudCBPIGFzIE9rdGFcblxuICBXLT4-UzogR2V0IHVzZXIgaW5mb3JtYXRpb25cbiAgTm90ZSBvdmVyIFcsUzogR0VUIC9tZSB3aXRoIGFjY2VzcyB0b2tlbiBhcyBCZWFyZXJcbiAgUy0tPj5POiBWYWxpZGF0ZSB1c2VyIGFjY2VzcyB0b2tlblxuICBOb3RlIG92ZXIgTyxTOiB2ZXJpZnkudmVyaWZ5QWNjZXNzVG9rZW4oPGJyLz50b2tlbiwgYXBwbGljYXRpb25OYW1lPGJyLz4pXG4gIE8tPj5TOiBSZXR1cm4gT2t0YSBhY2Nlc3MgY2xhaW1zXG4gIE5vdGUgb3ZlciBPLFM6IHsgLi4uLCB1aWQ6IHVzZXJJZCwgLi4uIH1cbiAgUy0tPj5POiBHZXQgdXNlciBwcm9maWxlXG4gIE5vdGUgb3ZlciBPLFM6IG9rdGFDbGllbnQuZ2V0VXNlcih1c2VySWQpXG4gIE8tPj5TOiBSZXR1cm4gdXNlciBwcm9maWxlXG4gIE5vdGUgb3ZlciBPLFM6IHsgPGJyIC8-Li4uLDxici8-dWlkOiB1c2VySWQsPGJyIC8-ZGlzcGxheU5hbWU6IGRpc3BsYXlOYW1lLDxiciAvPnN0YXR1czogJ0FDVElWRScsPGJyIC8-Li4uIDxiciAvPn1cbiAgUy0-PlM6IFJldHJpZXZlIGF1dGhvcml6YXRlZCByb2xlcyBmcm9tIGRhdGFiYXNlXG4gIFMtPj5TOiBSZXRyaXZlIGFsbG93ZWQgYWN0aXZpdGllcyBmcm9tIGRhdGFiYXNlXG4gIFMtLT4-VzogUmV0dXJuIHVzZXIgcHJvZmlsZSwgcm9sZSwgYW5kIGFjY2Vzc1xuICBOb3RlIG92ZXIgVyxTOiB7PGJyLz4uLi4sPGJyPmFjdGl2aXRpZXM6IFthY3Rpdml0aWVzXSw8YnIvPmlkOiB1c2VySWQsPGJyLz4gbmFtZTogZGlzcGxheU5hbWUsPGJyLz5zdGF0ZTogc3RhdGUsPGJyLz5yb2xlOiByb2xlLDxici8-Li4uPGJyLz59IiwibWVybWFpZCI6e30sInVwZGF0ZUVkaXRvciI6ZmFsc2V9

Comment (thetif) [reply documentation/sequence-diagram-authorization.md:4]: The frontend communicates directly with Okta though and then once the person is authenticated they communicate with our server. Our server does communicate with Okta too, but that's only for authorization not authentication

Comment (thetif) [reply documentation/sequence-diagram-authorization.md:4]: Disregard previous, yes, we can change this. :D

File: web/src/containers/LoginPageRoutes.test.js
Comment (knollfear) [inline web/src/containers/LoginPageRoutes.test.js:123]: This test and the one above it seem to be duplicates in everything except name.

File: web/src/containers/LoginPageRoutes.test.js
Comment (knollfear) [inline web/src/containers/LoginPageRoutes.test.js:55]: There seems to be a lot of repetition in this test.  I think we could shorten it up considerably if we implemented a table test.

File: web/src/containers/admin/ManageUserTable.test.js
Comment (knollfear) [inline web/src/containers/admin/ManageUserTable.test.js:51]: This seems like another case for the <Foo {...defaultProps} { ...props} /> pattern and maybe a table test.  I don't want to hold this up though.  Do we do Tech debt tickets?

File: web/src/containers/admin/StateAdmin.test.js
Comment (knollfear) [inline web/src/containers/admin/StateAdmin.test.js:189]: should these just be deleted?  Why keep them?  Will they be turned on in the future?

File: web/src/reducers/auth.test.js
Comment (knollfear) [inline web/src/reducers/auth.test.js:207]: Is this right?   It doesn't match the title.

**Review by knollfear — COMMENTED** [review]
I left a few comments.  I think they can all be fixed in another ticket except for a test that seems to have an incorrect name.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (tbolt) [inline .github/PULL_REQUEST_TEMPLATE.md]: is this supposed to be Developer?

File: terraform/test/test.tf
Comment (tbolt) [inline terraform/test/test.tf]: this test file staying in?

**Review by tbolt — APPROVED** [review]
Seems fine. It was tough to parse which code is specific to the refactor but I think I reviewed most of that code and it looks good.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (thetif) [reply .github/PULL_REQUEST_TEMPLATE.md]: probably 😳

Comment (thetif) [reply .github/PULL_REQUEST_TEMPLATE.md]: I'll update that next week when we make the changes to the templates

**Review by eStar386 — COMMENTED** [review]
https://github.com/CMSgov/eAPD/issues/3007

**Review by beparticular — APPROVED** [review]
I’ve poked around at this PR with a few log ins and i see where there is no cancel button anymore, i see the new new back to log in link, i see the combined authenticator stuff. not getting any weird log out issues. seeing the new locked out screen. I think that is everything I need to check, but if I missed anything please let me know and I'll take another look!

**Review by NAretakis — APPROVED** [review]
Login looking snappy and also working 😉 . Good to go.

**Pull request merged**
